guides/release/in-depth-topics/patterns-for-components.md

```diff
@@ -117,134 +127,191 @@ The most frequently used of these is `aria-describedby` and `aria-labelledby`.
+Below is an example of choosing different components for
+displaying different kinds of posts.  First, define your two components:
 
-The first parameter of the helper is the name of a component to render, as a
-string. So `{{component 'blog-post'}}` is the same as using `<BlogPost />`.
 
-The real value of [`{{component}}`](https://api.emberjs.com/ember/release/classes/Ember.Templates.helpers/methods/component?anchor=component)
-comes from being able to dynamically pick the component being rendered. Below is
-an example of using the helper as a means of choosing different components for
-displaying different kinds of posts:
+```gjs {data-filename=app/components/root-post.gjs}
+<template>
+  <h3>Hello from root!</h3>
+  <p>{{@post.body}}</p>
+</template>
+```
 
-```handlebars {data-filename=app/components/foo-component.hbs}
-<h3>Hello from foo!</h3>
-<p>{{this.post.body}}</p>
+```gjs {data-filename=app/components/reply-post.gjs}
+<template>
+  <h3>Hello from reply!</h3>
+  <div>{{@post.author}}</div>
+</template>
 ```
 
-```handlebars {data-filename=app/components/bar-component.hbs}
-<h3>Hello from bar!</h3>
-<div>{{this.post.author}}</div>
+Then, you can choose which to render based on the string data. We can pass the `@post` into our `ShowPostComponent` and it can render the correct type for us: 
+
+```gjs {data-filename=app/components/show-post.gjs}
+import Component from '@glimmer/component';
+
+import RootPost from  'my-app/components/root-post';
+import ReplyPost from 'my-app/components/reply-post';
+
+
+export default class ShowPostComponent extends Component {
+  // returns either RootPost or ReplyPost (default: RootPost)
+  get postComponent(postType) {
+    return this.args.post?.postType === 'reply' ? ReplyPost : RootPost;
+  }
+
+  <template>
+    <this.postComponent @author={{@post.author}} @body={{@post.body}} />
+  </template>
+}
 ```
 
-```handlebars {data-filename=app/templates/index.hbs}
-{{#each this.myPosts as |post|}}
-  {{!-- either foo-component or bar-component --}}
-  {{component post.postType post=post}}
-{{/each}}
+This is great when `RootPost` and `ReplyPost` take the same arguments, like `author` and `body` in the above example. But what if the components accept different arguments? One way would be to move the selection conditional into the template, like so:
+
+```gjs {data-filename=app/components/show-post.gjs}
+import RootPost from 'my-app/components/root-post';
+import ReplyPost from 'my-app/components/reply-post';
+
+const eq = (a, b) => a === b;
+
+<template>
+  {{!-- Post is either RootPost or ReplyPost --}}
+  {{#if (eq @post.postType "reply")}}
+    <ReplyPost
+      @author={{post.author}}
+      @body={{post.body}}
+      @replyTo={{post}}
+    />
+  {{else}}
+    <RootPost
+      @author={{post.author}}
+      @body={{post.body}}
+      @category={{post.topic}}
+    />
+  {{/if}}
+</template>
 ```
 
-or
+## Contextual Components
```

Comment:
aka currying, aka partial application

see also:
- helper
- modifier
- fn

(all do the same thing, for their respective invokables)